packages/core/src/client.ts

```diff
@@ -692,17 +693,36 @@ export class MedplumClient extends EventTarget {
 
     if (options?.accessToken) {
       this.setAccessToken(options.accessToken);
+      this.initPromise = Promise.resolve();
+    } else if ((this.storage as AsyncBackedClientStorage).initialized !== undefined) {
+      const initPromise = (this.storage as AsyncBackedClientStorage).initialized;
+      initPromise
+        .then(() => {
+          this.attemptResumeActiveLogin().catch(console.error);
+        })
+        .catch(console.error);
+      this.initPromise = initPromise;
     } else {
-      const activeLogin = this.getActiveLogin();
-      if (activeLogin) {
-        this.setAccessToken(activeLogin.accessToken, activeLogin.refreshToken);
-        this.refreshProfile().catch(console.log);
-      }
+      this.attemptResumeActiveLogin().catch(console.error);
+      this.initPromise = Promise.resolve();
     }
 
     this.setupStorageListener();
   }
 
+  get initialized(): Promise<void> {
```

Comment:
Oof.  I understand the need for this, but we need a better name and heavy docs.

Can we make it a normal method called `getInitPromise()`?

Reply:
Yes 100%, was planning to document before merging.

I also think I understand your fear of having a `Promise` on the object rather than just calling a getter function. I've seen this pattern using getters more these days but I guess it still may be unintuitive to await a field on an object for some users.